scheduler/scheduler.go

```diff
@@ -107,6 +117,111 @@ type managesWork interface {
 	Work(job) queuedJob
 }
 
+// Implemented as a separate function so that defer calls
+// are properly handled and cleanup done properly
+// (like for the removal of temporary Yaml to JSON conversion)
+func autoDiscoverTasks(taskFiles []os.FileInfo, fullPath string,
+	fp func(sch schedule.Schedule,
+		wfMap *wmap.WorkflowMap,
+		startOnCreate bool,
+		opts ...core.TaskOption) (core.Task, core.TaskErrors)) {
+	// Note that the list of files is sorted by name due to ioutil.ReadDir
+	// default behaviour. See go doc ioutil.ReadDir
+	for _, file := range taskFiles {
+		f, err := os.Open(path.Join(fullPath, file.Name()))
+		if err != nil {
+			log.WithFields(log.Fields{
+				"_block":           "autoDiscoverTasks",
+				"_module":          "scheduler",
+				"autodiscoverpath": fullPath,
+				"task":             file.Name(),
+			}).Error("Opening file ", err)
+			continue
+		}
+		defer f.Close()
+		if !strings.HasSuffix(file.Name(), ".json") {
+			fc, err := ioutil.ReadAll(f)
+			if err != nil {
+				log.WithFields(
+					log.Fields{
+						"_block":           "autoDiscoverTasks",
+						"_module":          "scheduler",
+						"autodiscoverpath": fullPath,
+						"task":             file.Name(),
+					}).Error("Reading Yaml file ", err)
+				continue
+			}
+			js, err := yaml.YAMLToJSON(fc)
+			if err != nil {
+				log.WithFields(
+					log.Fields{
+						"_block":           "autoDiscoverTasks",
+						"_module":          "scheduler",
+						"autodiscoverpath": fullPath,
+						"task":             file.Name(),
+					}).Error("Parsing Yaml file ", err)
+				continue
+			}
+			tfile, err := ioutil.TempFile(os.TempDir(), "yaml2json")
```

Comment:
@IRCody I do not understand fully what you are suggesting here. CreateTaskFromContent is using json library to decode task informations, and if I pass a Yaml file directly to it I get 

`ERRO[0001] invalid character '-' in numeric literal      _block=autoDiscoverTasks _module=scheduler autodiscoverpath=/home/olivier/.gvm/pkgsets/go1.5.4/global/src/github.com/intelsdi-x/snap/build/plugin task=mock-file.yaml
`

So I do not see how to get around this by not having a temp file and keeping the code as common as possible to what it was (not replacing json by yaml in CreateTaskFromContent.

Thanks for your inputs